### PR TITLE
Disable 'Prefill with' if no years or same year only

### LINKED ISSUE #2629: Original Data Point: Disable prefill select if no other ODPs
image

### LINKED ISSUE #2865: Table 1a / 1b select a year to prefill data - sort years
@minotogna @sorja 

The list of years should follow the numeric order. For example: 1990, 1995, 2000; not 1990, 2000, 1995.

### PR DESCRIPTION
**Pull request by sorja**
Resolves #2865 

If no year or only current editing year

image

### CODE REVIEW

File: src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx [hunk 1]
```diff
@@ -34,11 +34,12 @@ const NationalClasses: React.FC<Props> = (props) => {
   const { print } = useIsPrint()
   const originalDataPointUpdating = useIsOriginalDataPointUpdating()
   const reservedYearsWithClasses = (useOriginalDataPointReservedYears() ?? [])
+    .filter((reservedYear) => Number(reservedYear.year) !== Number(year))
     .filter((reservedYear) => reservedYear.nationalClasses > 0)
     .map((reservedYear) => reservedYear.year)
```
Comment (minotogna) [inline src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx]: filter + filter + map => better one reduce ?

File: src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx [hunk 2]
```diff
@@ -1,126 +1,24 @@
...
-                  canEditData={canEditData}
-                />
-              ))}
-            </tbody>
-          </table>
-        </div>
-      </div>
+      <Title />
+      <Prefill canEditData={canEditData} originalDataPoint={originalDataPoint} />
+      <NationalClassesTable canEditData={canEditData} originalDataPoint={originalDataPoint} />
```
Comment (sorja) [inline src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx:21]: Moved to components

File: src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx [hunk 3]
```diff
@@ -1,126 +1,24 @@
...
-  // Copying is disabled if: odp doesn't have a year, there is no previous years or previous year is not selected
-  const copyDisabled = !year || year === -1 || originalDataPointUpdating
-
-  const onCopyClick = useCallback(() => {
-    if (window.confirm(t('nationalDataPoint.confirmCopyPreviousValues'))) {
-      dispatch(
-        OriginalDataPointActions.copyPreviousNationalClasses({
-          originalDataPoint,
-          assessmentName: assessment.props.name,
-          cycleName: cycle.name,
```
Comment (sorja) [inline src/client/pages/OriginalDataPoint/components/NationalClasses/NationalClasses.tsx:49]: logic moved to hooks (only used by components)

File: src/client/pages/AssessmentPrint/OriginalDataPointsPrint/useOriginalDataPoints.ts [hunk 4]
```diff
@@ -27,5 +27,13 @@ export const useOriginalDataPoints = (): { originalDataPoints: Array<OriginalDat
     fetchResults()
   }, [])
 
-  return { originalDataPoints, loading }
+  const originalDataPoints = useMemo(
+    () => data.sort((a: OriginalDataPoint, b: OriginalDataPoint) => a.year - b.year),
```
Comment (sorja) [inline src/client/pages/AssessmentPrint/OriginalDataPointsPrint/useOriginalDataPoints.ts:31]: fix minor issue with odp order in printview

**Review by minotogna — APPROVED** [review]
that's a good and nice refactor 👍 🤩 
next time a video , please ;) 🙏

**Pull request merged**